articles/quickstart/webapp/php/01-login.md

```diff
@@ -11,159 +12,214 @@ topics:
+Route::add('/logout', 'onLogoutRoute');
+
+// This tells our router that we've finished configuring our routes, and we're ready to begin routing incoming HTTP requests:
+Route::run('/');
 ```
 
-```php
-// logout.php
-
-// ...
-$auth0->logout();
-$return_to = 'http://' . $_SERVER['HTTP_HOST'];
-$logout_url = sprintf('http://%s/v2/logout?client_id=%s&returnTo=%s', '${account.namespace}', '${account.clientId}', $return_to);
-header('Location: ' . $logout_url);
-die();
+Let's also define our sample application's route's full URLs as named constants for convenience sake, as we'll need to reference them in a few places through our sample application:
+
+```PHP
+// 👆 We're continuing from the steps above. Append this to your index.php file.
+
+define('ROUTE_URL_INDEX', rtrim($_ENV['AUTH0_BASE_URL'], '/'));
+define('ROUTE_URL_LOGIN', ROUTE_URL_INDEX . '/login');
+define('ROUTE_URL_CALLBACK', ROUTE_URL_INDEX . '/callback');
+define('ROUTE_URL_LOGOUT', ROUTE_URL_INDEX . '/logout');
 ```
 
-### Optional: Configure session data
+Now we can move on to adding our application's route handling logic, and the SDK integrations:
 
-By default, the SDK stores user information in the PHP session and discards the access and ID Tokens.
+### Checking for a session
 
-To keep the tokens, to the SDK configuration, pass the following:
-* `'persist_access_token' => true`
-* `'persist_id_token' => true`
+The Auth0 PHP SDK has a convenient method for checking if our user has authenticated and returned their profile, `getCredentials()`. Let's install this on our index route to print the user profile if they're logged in or report that they need to login.
 
-To disable the session, pass `'store' => false` to the SDK configuration.
+```PHP
+// 👆 We're continuing from the steps above. Append this to your index.php file.
 
-Instead of using the PHP session to store information, you can use Laravel, Zend, Symfony or similar techniques. To do that, create a class that implements the get, set and delete methods and pass it to the SDK.
+function onIndexRoute() {
+  $session = $auth0->getCredentials();
 
-```php
-// index.php
-
-$laravelStore = new MyLaravelStore();
-$auth0 = new Auth0(array(
-    // ...
-    'store' => $laravelStore,
-    // ...
-));
+  if ($session === null) {
+    // The user isn't logged in.
+    echo 'Please <a href="/login">log in</a>.</p>';
+    return;
+  }
+
+  // The user is logged in.
+  print_r($session->user);
+
+  echo 'You can now <a href="/logout">log out</a>.</p>';
+}
+```
+
+We can access all the properties of our user's profile from the `user` property response, which is an array. So, for example, we can pull the user's nickname using `$session->user['nickname']`, or their email address from `$session->user['email']`. The structure is a [normalized user profile](https://auth0.com/docs/users/user-profile-structure), which you can learn more about [here](https://auth0.com/docs/users/normalized-user-profiles).
```

Comment:
I know `nickname` it's an open id profile property, but I'm not sure if it's often populated. Consider using a different one, like `given_name` or just `name`.

Also, bear in mind that `email` will only be available if the scope included "email" in it.